### PR TITLE
[REVIEW] Add support for `pipe` API

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #4102

This PR adds support for `pipe` API in `Series/DataFrame/Index` and `GroupBy`.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/utils/utils.py [hunk 1]
```diff
@@ -394,3 +394,16 @@ def pa_mask_buffer_to_mask(mask_buf, size):
         dbuf.copy_from_host(np.asarray(mask_buf).view("u1"))
         return Buffer(dbuf)
     return Buffer(mask_buf)
+
+
+def _pipe(obj, func, *args, **kwargs):
```
Comment (shwina) [inline python/cudf/cudf/utils/utils.py:399]: Should we just create a `core.common` similar to Pandas for things like this? I wouldn't consider `_pipe` to be a "utility".

Comment (galipremsagar) [reply python/cudf/cudf/utils/utils.py]: Sounds good to me, created `cudf.core.common` 👍

**Review by shwina — APPROVED** [review]
Nice work!

**Pull request merged**